### PR TITLE
11 implement all messages needed for frbc control type

### PR DESCRIPTION
**Pull request by TNO-SlaFleur**

### CODE REVIEW

**Comment by TNO-SlaFleur** [comment]
I got really bored writing unit tests so I rather made a script to generate the basic unit test cases. I also used a template generator to help me with the frbc messages which I also added. They are not extremely smart so manual intervention is needed but it saved me loads of time.

File: src/s2wsjson/common/number_range.py [hunk 1]
```diff
@@ -16,3 +16,12 @@ def validate_start_end_order(cls, values: dict[str, Any]) -> dict[str, Any]:
             raise ValueError(cls, 'start_of_range should not be higher than end_of_range')
 
         return values
+
+    def __hash__(self):
+        return hash(f'{self.start_of_range}|{self.end_of_range}')
+
+    def __eq__(self, other):
+        if isinstance(other, NumberRange):
+            return self.start_of_range == other.start_of_range and self.end_of_range and other.end_of_range
```
Comment (victorgarcia98) [inline src/s2wsjson/common/number_range.py:25]: A shorter form would be `return other.__hash__() == self.__hash__()`

File: src/s2wsjson/common/support.py
Comment (victorgarcia98) [inline src/s2wsjson/common/support.py:1]: This comes very handy!

File: src/s2wsjson/validate_values_mixin.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def to_json(self: C) -> str:
         try:
             return self.json(by_alias=True, exclude_none=True)
         except (ValidationError, TypeError) as e:
-            raise S2ValidationError(self, 'Pydantic raised a format validation error.', pydantic_validation_error=e)
+            raise S2ValidationError(self, 'Pydantic raised a format validation error.') from e
```
Comment (victorgarcia98) [inline src/s2wsjson/validate_values_mixin.py:66]: nice!!

File: tests/unit/frbc/frbc_fill_level_target_profile_test.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+from s2wsjson.common import *
+from s2wsjson.frbc import *
+
+
+class FRBCFillLevelTargetProfileTest(TestCase):
+    def test__from_json__happy_path_full(self):
+        # Arrange
+        json_str = """
+{
```
Comment (victorgarcia98) [inline tests/unit/frbc/frbc_fill_level_target_profile_test.py:14]: I think we should add a couple of tabs in this (or more general in the test generation script), if it's not too much of a change.

File: tests/unit/frbc/frbc_fill_level_target_profile_test.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            FRBCFillLevelTargetProfileElement(duration=Duration.from_timedelta(timedelta(milliseconds=4704)),
+                                              fill_level_range=NumberRange(end_of_range=10800.98606857073545,
+                                                                           start_of_range=6891.19014440217))])
+        self.assertEqual(frbc_fill_level_target_profile.message_id, uuid.UUID("04a6c8af-ca8d-420c-9c11-e96a70fe82b1"))
+        self.assertEqual(frbc_fill_level_target_profile.message_type, "FRBC.FillLevelTargetProfile")
+        self.assertEqual(frbc_fill_level_target_profile.start_time,
+                         datetime(year=2021, month=4, day=17, hour=0, minute=19, second=20,
+                                  tzinfo=offset(offset=timedelta(seconds=0.0))))
+
+    def test__to_json__happy_path_full(self):
```
Comment (victorgarcia98) [inline tests/unit/frbc/frbc_fill_level_target_profile_test.py:44]: This test maybe requires manual styling or running something like black.

File: tests/unit/frbc/frbc_usage_forecast_element_test.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+from s2wsjson.common import *
+from s2wsjson.frbc import *
+
+
+class FRBCUsageForecastElementTest(TestCase):
+    def test__from_json__happy_path_full(self):
+        # Arrange
+        json_str = """
+{
```
Comment (victorgarcia98) [inline tests/unit/frbc/frbc_usage_forecast_element_test.py:14]: From this test, I noticed that the percentiles values are not being tested/generated properly:

```
lower limit <= 96 lower limit <= 68 lower limit <= 68 upper limit <= 95 upper limit <= upper limit
```

We can tackle this is a separate ticket.

**Review by victorgarcia98 — APPROVED** [review]
Superb work!

File: src/s2wsjson/common/number_range.py (hunk 1 above)
Comment (TNO-SlaFleur) [reply src/s2wsjson/common/number_range.py:25]: While I agree it is definitely shorter and does not require any changes in the future if new fields are added (however unlikely), there is a performance consideration. The hash implementation requires generating a new string. If each `==` first transforms any NumberRange to a `str`,  I expect it to be a lot slower than the current condition which checks both floats on equality.

I am unsure if we should treat performance considerations heavily at the moment. What do you think?

Comment (TNO-SlaFleur) [reply src/s2wsjson/common/number_range.py:25]: Btw, I just noticed a bug. The `end_of_range` is not checked for equality. Fixing now xD

File: tests/unit/frbc/frbc_fill_level_target_profile_test.py (hunk 4 above)
Comment (TNO-SlaFleur) [reply tests/unit/frbc/frbc_fill_level_target_profile_test.py:44]: Definitely would like black to take care of this.

Comment (TNO-SlaFleur) [reply tests/unit/frbc/frbc_fill_level_target_profile_test.py:44]: Added as #23

File: tests/unit/frbc/frbc_usage_forecast_element_test.py (hunk 5 above)
Comment (TNO-SlaFleur) [reply tests/unit/frbc/frbc_usage_forecast_element_test.py:14]: See #24

File: tests/unit/frbc/frbc_fill_level_target_profile_test.py (hunk 3 above)
Comment (TNO-SlaFleur) [reply tests/unit/frbc/frbc_fill_level_target_profile_test.py:14]: As it is a multiline string, any tabs will show up in the `json_str`. Json loader should not mind it but it does change the value.
Fixing in seperate ticket: #25

File: src/s2wsjson/common/number_range.py (hunk 1 above)
Comment (TNO-SlaFleur) [reply src/s2wsjson/common/number_range.py]: Fixed!

**Pull request merged**